src/groups/semidirect_product_group.jl

```diff
@@ -224,11 +224,17 @@ function _semidirect_maybe_inv(::GroupAction{<:AbstractLeftGroupActionType}, G,
     return copy(G, g)
 end
 # and in in-place
-function _semidirect_maybe_inv!(::GroupAction{<:AbstractRightGroupActionType}, G, k, g)
-    return inv!(G, k, g)
-end
-function _semidirect_maybe_inv!(::GroupAction{<:AbstractLeftGroupActionType}, G, k, g)
-    return copyto!(G, k, g)
+# If AO and A are "aligned", we have to invert, otherwise we do not
+function _semidirect_maybe_inv!(::Type{AO}, ::GroupAction{A}, G, k, g) where {AO, A}
+    if (A <: AbstractLeftGroupActionType) != (AO == ActionActsOnLeft)
+        # act on left && right action ||
+        # act on right && left action
+        return copyto!(G, k, g)
+    else
+        # act on left && left action ||
+        # act on right && right action
+        return inv!(G, k, g)
+    end
```

Comment:
Previously this was done via dispatch. Now it is an if. Why?

Reply:
I found it easier to understand with one function and an if statement than with 4 when using dispatch.

Reply:
But the if statement is also very cryptic. Is is basically if we have a left group action _and_ it acts on right – as well as we have a right group action and it acts on left?
Why is the first a `<:` and the second a `==`?

Maybe `(A <: AbstractLeftGroupAction) == (AO <: ActionActsOnLeft)` (and switch your two blocks of course) Is easier to parse (for humans)?

It still for 100% changes the logic, since the old one had one case for left actions one for right actions. So if the docs are not updated – they are wrong.

Reply:
Look at the comments in each case.
```julia
if  act_on_left && right_action || act_ on_right && left_action 
    return copyto!(G, k, g)
elseif  act_on_left && left_action || act_on_right && right_action
    return inv!(G, k, g)
end
```
This is the core of the change of when to call inverse, it doesn't only depend on `act_on_left` or `act_on_right`, but the combination.